ui/src/Queues/RefreshQueuesData.ts

```diff
@@ -0,0 +1,70 @@
+import {Dispatch, SetStateAction, useEffect, useState} from "react";
+import QueueService from "../services/QueueService";
+import {QueueMessagesData, QueueStatistic} from "./QueueMessageData";
+
+function useRefreshedQueueStatistics(): QueuesStatisticsDataControl {
+    function convertQueueStatisticsToNewQueueData(newQuery: QueueStatistic) {
+        return {
+            queueName: newQuery.name,
+            currentMessagesNumber: newQuery.statistics.approximateNumberOfVisibleMessages,
+            delayedMessagesNumber: newQuery.statistics.approximateNumberOfMessagesDelayed,
+            notVisibleMessagesNumber: newQuery.statistics.approximateNumberOfInvisibleMessages,
+            isOpened: false
+        } as QueueMessagesData
+    }
+
+    function updateNumberOfMessagesInQueue(newStatistics: QueueStatistic, knownQuery: QueueMessagesData) {
+        return {
+            ...knownQuery,
+            currentMessagesNumber: newStatistics.statistics.approximateNumberOfVisibleMessages,
+            delayedMessagesNumber: newStatistics.statistics.approximateNumberOfMessagesDelayed,
+            notVisibleMessagesNumber: newStatistics.statistics.approximateNumberOfInvisibleMessages,
+        }
+    }
+
+    async function getInitialStatistics() {
+        return QueueService.getQueueListWithCorrelatedMessages().then(QueuesStatistics => {
```

Comment:
Minor thing, QueuesStatistics -> queuesStatistics if there is no particular reason why it's made that way.

Reply:
Renamed